### ISSUE TITLE
No Sound played, No Badge displayed.

### ISSUE THREAD

**Issue by siddharth-kt** [OneSignal/react-native-onesignal#1338]
**Description:**

I am facing 2 problems.
- push-notification does not play sound or vibrate (for "high" importance android-channels.)
- Badge are not displayed at all on my App Icon. (even for "urgent" importance android-channels.)
  (I just noticed documentation does not allow badges for Vivo devices 
  (https://documentation.onesignal.com/docs/badges#android-badges), how to add support for badges on vivo devices ?)

I am using python for pushing notifications. (create-notification API)
my-python-code :

```py
import requests
import json

header = {"Content-Type": "application/json; charset=utf-8",
          "Authorization": "MY-REST-API-KEY"}

payload = {"app_id": "MY-APP-ID",
           "included_segments": ["Subscribed Users"],
           "data": {"abc": 123},
           "android_channel_id": "MY-CHANNEL-ID",
           "large_icon": "https://source.unsplash.com/collection/29",
           "contents": {"en": "Commented on your post"},
           "headings": {"en": "sid"},
            "buttons": [
                {"id": "id2", "text": "like"},
                {"id": "id1", "text": "reply"}
            ]}
 
req = requests.post("https://onesignal.com/api/v1/notifications", headers=header, data=json.dumps(payload))
print(req.status_code, req.reason)
```

**Environment**
Android: 11
react-native: 0.65.1
react-native-onesignal: 4.3.3

- I installed package using npm

**Steps to Reproduce Issue:**
1.) Install the OneSignal SDK using npm into your project
2.) Try running same python code as defined above.
3.) Attempt to receive a push notification.

I edited my question a bit because I observed that previous issue(s) occurs only some times.

**Comment by DenisDov** [comment]
Same on ios, no sound, no badge, no notification visible, only vibration is triggered...

**Comment by dombartenope** [comment]
Hey there @siddharth-kt & @DenisDov 

I attempted to recreate this issue on my end using React Native .65.1 and the OneSignal React Native plugin 4.3.3 on a few different Android devices (11) but am unable to reproduce this with your python request. 

As for IOS, I also did not receive badges upon making the request but was alerted under every android_channel_id condition and excluding it all together.

![IMG_4781 (1)](https://user-images.githubusercontent.com/56173293/151241631-030ee63a-5c28-4bcb-b793-46ab02ad49ec.jpg)
![IMG_4782 (1)](https://user-images.githubusercontent.com/56173293/151241633-d1af36ff-7d1f-4621-bc96-f0cebed317fb.jpg)

Are either of you noticing this happen when sending a notification out from the dashboard as well, or specifically when making a direct request to the api?

For android: I am unable to test this on a vivo device myself, but it would appear that this could be at the device level, as I am not seeing it occur on multiple devices at this point. There are a few resources online, regarding notifications with vivo behaving oddly, that might be worth looking into.

Feel free to reach out to support@onesignal.com for quicker visibility by the support team, and open a ticket with us, and we'd be happy to troubleshoot this further with you.

Thanks!
